modules/axtask/src/task.rs

```diff
@@ -39,9 +46,28 @@ pub struct TaskInner {
     entry: Option<*mut dyn FnOnce()>,
     state: AtomicU8,
 
+    /// CPU affinity mask.
+    cpumask: SpinNoIrq<CpuSet>,
+
+    /// Mark whether the task is in the wait queue.
     in_wait_queue: AtomicBool,
+    /// Mark whether the task is in the timer list.
     #[cfg(feature = "irq")]
     in_timer_list: AtomicBool,
+    /// A ticket ID used to identify the timer event.
+    /// Incremented by 1 each time the timer event is triggered or expired.
+    #[cfg(feature = "irq")]
+    timer_ticket_id: AtomicUsize,
```

Comment:
No, we don't think we can simply use `timer_ticket_id == 0` to indicate a task is not in time list.

`timer_ticket_id` means current availble timer id, it has to be incremented each time a timer is triggered or expired.

If we set `timer_ticket_id` to 0, we cannot determine whether the previous timer event for this task was triggered by the timer or by a notification.

* The timer ticket ID is used to address this issue: the original timer event has been canceled, but before the deadline is reached, a new timer event occurs, setting in_timer_list to true. When the deadline for the original timer event arrives, it will be triggered.
*  The reason I keep the `in_timer_list` flag is to determine in the wait queue's `cancel_events`, whether to call `timer_ticket_expire_one`, instead of simply incrementing the timer ticket ID every time in `cancel_events`.

Reply:
see `modules/axtask/src/wait_queue.rs`
```Rust
    fn cancel_events(&self, curr: CurrentTask) {
        // A task can be wake up only one events (timer or `notify()`), remove
        // the event from another queue.
        if curr.in_wait_queue() {
            // wake up by timer (timeout).
            let mut wq_locked = self.queue.lock();
            wq_locked.retain(|t| !curr.ptr_eq(t));
            curr.set_in_wait_queue(false);
        }
        #[cfg(feature = "irq")]
        if curr.in_timer_list() {
            curr.set_in_timer_list(false);
            curr.timer_ticket_expire_one();
            // TODO:
            // this task is still not removed from timer list of target CPU,
            // which may cause some redundant timer events.
        }
    }
```